compiler/src/dotty/tools/dotc/transform/TreeTransform.scala

```diff
@@ -210,9 +210,16 @@ object TreeTransforms {
    */
   class NXTransformations {
 
-    private def hasRedefinedMethod(cls: Class[_], name: String): Boolean =
-      if (cls.getDeclaredMethods.exists(_.getName == name)) cls != classOf[TreeTransform]
-      else hasRedefinedMethod(cls.getSuperclass, name)
+    private def hasRedefinedMethod(cls: Class[_], name: String): Boolean = {
+      val clsMethods = cls.getDeclaredMethods
+      var i = clsMethods.length - 1
+      while (i >= 0) {
+        if (clsMethods(i).getName == name)
+          return cls != classOf[TreeTransform]
+        i -= 1
+      }
+      hasRedefinedMethod(cls.getSuperclass, name)
+    }
```

Comment:
I wonder why this method shows up as hot. It should only be called during initialization when we compute the tree transformers. How often is the method called? Back of the envelope calculation: 100 phases * 50 methods, i.e. 5000? If its significantly more than that there's something we have overlooked.